pstatus-report-sink-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/plugins/AWSSQSProcessor.kt

```diff
@@ -1,62 +1,78 @@
 package gov.cdc.ocio.processingstatusapi.plugins
 
 import com.google.gson.JsonSyntaxException
+
+import java.util.*
+
 import gov.cdc.ocio.processingstatusapi.exceptions.BadRequestException
 import gov.cdc.ocio.processingstatusapi.exceptions.BadStateException
 import gov.cdc.ocio.processingstatusapi.models.CreateReportMessage
 import gov.cdc.ocio.processingstatusapi.models.Source
+import gov.cdc.ocio.processingstatusapi.models.ValidationComponents
 import gov.cdc.ocio.processingstatusapi.utils.SchemaValidation
-import gov.cdc.ocio.processingstatusapi.utils.SchemaValidation.Companion.gson
-import gov.cdc.ocio.processingstatusapi.utils.SchemaValidation.Companion.logger
+import gov.cdc.ocio.reportschemavalidator.service.SchemaValidationService
 
 /**
  * The AWS SQS service is an additional interface for receiving and validating reports.
  */
 class AWSSQSProcessor {
-    /**
```

Comment:
There is a lot of duplicate code in the AWSSQSProcessor, RabbitMQProcessor, and ServiceBusProcessor.  Please consider making a base class called, MessageProcessor that all three of these classes derive from.  Move the common functionality into the base class and invoke the base class method to validateMessage().

Reply:
I just committed fix for this. Please take a look and let me know your thoughts!